src/main/java/org/jitsi/jigasi/lobby/Lobby.java

```diff
@@ -0,0 +1,259 @@
+
+        super.leaveRoom();
+    }
+
+    public void addLobbyListener(LobbyListener lobbyListener)
+    {
+        if (this.lobbyListeners != null)
+        {
+            this.lobbyListeners.add(lobbyListener);
+        }
+    }
+
+    public void removeLobbyListener(LobbyListener lobbyListener)
+    {
+        if (this.lobbyListeners != null)
+        {
+            this.lobbyListeners.remove(lobbyListener);
+        }
+    }
+
+    private void approve(ChatRoomInvitationReceivedEvent chatRoomInvitationReceivedEvent)
+    {
+        if (this.lobbyListeners != null)
+        {
+            for (LobbyListener ll: this.lobbyListeners)
+            {
+                ll.onLobbyAccessApproved(chatRoomInvitationReceivedEvent, this);
+            }
+        }
+    }
+
+    private void reject()
+    {
+        if (this.lobbyListeners != null)
+        {
+            for (LobbyListener ll : this.lobbyListeners)
+            {
+                ll.onLobbyAccessRejected(this);
+            }
+        }
+    }
+
+    @Override
+    public void invitationReceived(ChatRoomInvitationReceivedEvent chatRoomInvitationReceivedEvent)
+    {
+        this.approve(chatRoomInvitationReceivedEvent);
+    }
+
+    @Override
+    public void localUserPresenceChanged(LocalUserChatRoomPresenceChangeEvent localUserChatRoomPresenceChangeEvent)
+    {
+        if (localUserChatRoomPresenceChangeEvent.getEventType() == LocalUserChatRoomPresenceChangeEvent.LOCAL_USER_KICKED)
+        {
+            this.reject();
+        }
+    }
+
+    public interface LobbyListener
+    {
+        void onLobbyAccessApproved(ChatRoomInvitationReceivedEvent chatRoomInvitationReceivedEvent, Lobby lobby);
+        void onLobbyAccessRejected(Lobby lobby);
+    }
+
+    @Override
+    protected CallContext getCallContext()
+    {
+        return this.callContext;
+    }
+
+    protected Jid getRoomJid()
+    {
+        return this.roomJid;
+    }
+
+    @Override
+    protected byte[] getRoomPassword()
+    {
+        return new byte[]{};
+    }
+
+    @Override
+    protected ProtocolProviderService getProtocolProvider()
+    {
+        return this.xmppProvider;
+    }
+
+    @Override
+    public Localpart getResourceIdentifier()
+    {
+        return this.roomJid.getLocalpartOrNull();
+    }
+
+    @Override
+    protected void setupChatRoom(ChatRoom mucRoom)
+    {
+        if (mucRoom instanceof ChatRoomJabberImpl)
+        {
+            if (this.displayName != null)
+            {
+                ((ChatRoomJabberImpl)mucRoom).addPresencePacketExtensions(
+                        new Nick(this.displayName));
+            }
+            else
+            {
+                logger.error(this.callContext
+                        + " No display name to use...");
+            }
+
+            String region = JigasiBundleActivator.getConfigurationService()
+                    .getString(LOCAL_REGION_PNAME);
+            if(!StringUtils.isNullOrEmpty(region))
```

Comment:
We do not need this. This is for the main room so we inform jicofo for our region to select us appropriate jvb. We do not have media for lobby room so no need.

Reply:
Removed